src/Game/SingleMode.cpp

```diff
@@ -1,12 +1,17 @@
 #include "SingleMode.h"
 
-SingleMode::SingleMode() {}
+SingleMode::SingleMode(COMMON::EGameMode game_mode)
+{
+  m_mode_game = game_mode;
+}
```

Comment:
Конструктор не должен принимать каких либо параметров. Вместо это он просто устанавливает тип игры
```suggestion
SingleMode::SingleMode()
{
  m_mode_game = COMMON::EGameMode::SINGLE;
}
```